src/Ferrite.jl

```diff
@@ -32,12 +32,9 @@ struct RefPrism         <: AbstractRefShape{3} end
 # TODO: Update interpolation definitions and enable deprecation
 const RefCube = RefHexahedron
 
-"""
-Abstract type which has `CellValues` and `FaceValues` as subtypes
-"""
-abstract type Values{dim,T,refshape} end
-abstract type CellValues{dim,T,refshape} <: Values{dim,T,refshape} end
-abstract type FaceValues{dim,T,refshape} <: Values{dim,T,refshape} end
+abstract type AbstractValues{IP} end
+abstract type AbstractCellValues{IP} <: AbstractValues{IP} end
+abstract type AbstractFaceValues{IP} <: AbstractValues{IP} end
```

Comment:
I am pretty happy with the PR as is now, but I am not sure about this parametrization.

It seems that the interpolation is the only "common" thing between the various implementations, so it seems natural to use it. However, it is also possible to not have a type parameter at all. Internally this is only used to define https://github.com/Ferrite-FEM/Ferrite.jl/blob/7c5b517e450aff17beae66b5f732b44156742395/src/FEValues/common_values.jl#L5-L6 which is used for 3 or 4 dispatches.

Looking at the examples, in some cases the previous `dim, T` parameters where used to initialize some things, but might be better to provide `get(space|ref)dim` functions for these cases.